src/main/java/spoon/reflect/visitor/LiteralHelper.java

```diff
@@ -43,13 +44,19 @@ private static String getBasedString(Long value, LiteralBase base) {
 	}
 
 	private static String getBasedString(Float value, LiteralBase base) {
+		if (value.isInfinite() || value.isNaN()) {
```

Comment:
How about a parameterized test for these values? Should be very little effort to see that we raise whenever the value is infinite or NaN. A test would also make the intent of the PR a little bit more clear.